src/clj/com/climate/claypoole/impl.clj

```diff
@@ -13,26 +13,19 @@
 
 (ns com.climate.claypoole.impl
   "Implementation helper functions for Claypoole."
-  (:require
-    [clojure.core :as core])
-  (:import
-    [clojure.lang
-     IFn]
-    [com.climate.claypoole.impl
-     Prioritized
-     PriorityThreadpoolImpl]
-    [java.util
-     Collection
-     List]
-    [java.util.concurrent
-     ExecutionException
-     Executors
-     ExecutorService
-     Future
-     LinkedBlockingQueue
-     ThreadFactory
-     TimeoutException
-     TimeUnit]))
+  (:require [clojure.core :as core])
+  (:import [clojure.lang IFn]
+           [com.climate.claypoole.impl Prioritized PriorityThreadpoolImpl]
+           [java.util Collection List]
```

Comment:
@borkdude : the import to List (used as a type hint in deftype bellow) is marked as unused by clj-kondo. 
Please let me know if this is a known thing or I should report it as an issue.

Reply:
If I remove it, the compilation fails.

Reply:
@ieugen Probably a bug. Please file a repro (smallest possible) in an issue in clj-kondo. Thanks.

Reply:
Reported here https://github.com/clj-kondo/clj-kondo/issues/1572